src/util/metadata/side_metadata/helpers.rs

```diff
@@ -252,22 +281,41 @@ pub fn find_last_non_zero_bit_in_metadata_bits(
         return FindMetaBitResult::UnmappedMetadata;
     }
     let byte = unsafe { addr.load::<u8>() };
-    if let Some(bit) = find_last_non_zero_bit_in_u8(byte) {
-        if bit >= start_bit && bit < end_bit {
-            return FindMetaBitResult::Found { addr, bit };
-        }
+    if let Some(bit) = find_last_non_zero_bit_u8(byte, start_bit, end_bit) {
+        return FindMetaBitResult::Found { addr, bit };
     }
     FindMetaBitResult::NotFound
 }
 
-fn find_last_non_zero_bit_in_u8(byte_value: u8) -> Option<u8> {
-    if byte_value != 0 {
-        let bit = byte_value.trailing_zeros();
-        debug_assert!(bit < 8);
-        Some(bit as u8)
-    } else {
-        None
-    }
+macro_rules! impl_find_last_non_zero_bit {
+    ($int_ty: ty, $value: expr, $start: expr, $end: expr) => {{
+        let mask = match (1 as $int_ty).checked_shl(($end - $start) as u32) {
+            Some(shl) => (shl - 1) << $start,
+            None => <$int_ty>::MAX << $start,
+        };
+        let leading_zeroes = ($value & mask).leading_zeros();
+        if leading_zeroes >= <$int_ty>::BITS {
```

Comment:
We can just compare `$value & mask` with zero.  If not zero, there must be at least one non-zero bit.